lib/charms/operator_libs_linux/v1/systemd.py

```diff
@@ -0,0 +1,230 @@
+# Copyright 2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Abstractions for stopping, starting and managing system services via systemd.
+
+This library assumes that your charm is running on a platform that uses systemd. E.g.,
+Centos 7 or later, Ubuntu Xenial (16.04) or later.
```

Comment:
This is something that should be updated/discussed in canonical/operator-libs-linux where the systemd charm library is hosted. I'm only consuming this lib so that I can use it within `utils.slurmd` for controlling the slurmd service. I've already taken on some work to update the systemd charm library so we can pass more arguments to `service_start(...)` here https://github.com/canonical/operator-libs-linux/pull/106. I would like to pass the `--no-block` option to systemctl when we start slurmd so that we do not block on waiting for the Python wrapper to determine if the slurmd daemon has started correctly.

Perhaps we issue a pull request to operator-libs-linux if we feel that the documentation for the systemd charm library is too broad.